### ISSUE TITLE
Uncaught AssertionError: null == true

### ISSUE THREAD

**Issue by sadlerdavid** [atom/atom#5504]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 0.177.0
**System**: Mac OS X 10.10.2
**Thrown From**: Atom Core

### Stack Trace

Uncaught AssertionError: null == true

```
At assert.js:100

AssertionError: null == true
  at TLSSocket.socketOnData (_http_client.js:309:3)
  at TLSSocket.emit (events.js:116:17)
  at TLSSocket.Readable.read (_stream_readable.js:374:10)
  at TLSSocket.socketCloseListener (_http_client.js:230:10)
  at TLSSocket.emit (events.js:138:20)
  at TCP.close (net.js:477:12)

```

### Commands

```
     -2:01.7 core:copy (atom-text-editor.editor)
     -2:01.2 core:move-right (atom-text-editor.editor)
     -2:00.0 editor:newline (atom-text-editor.editor)
     -1:59.2 core:paste (atom-text-editor.editor)
 20x -1:57.7 core:move-left (atom-text-editor.editor)
     -1:53.6 core:backspace (atom-text-editor.editor)
     -1:33.9 core:save (atom-text-editor.editor)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-dark-ui",
      "atom-light-syntax"
    ],
    "disabledPackages": [
      "deprecation-cop"
    ]
  },
  "editor": {
    "fontFamily": "Ubuntu",
    "fontSize": 11,
    "showInvisibles": true,
    "showIndentGuide": true,
    "tabLength": 4,
    "softWrap": true,
    "useShadowDOM": true,
    "invisibles": {},
    "zoomFontWhenCtrlScrolling": true
  }
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.21.3
atom-django, v0.2.0
atom-python-debugger, v0.2.0
django-templates, v0.3.0
language-django, v0.1.1
language-python-django, v0.3.1
language-python-django-templates, v0.1.0
linter, v0.11.1
linter-python-pep8, v0.2.0
linter-python-pyflakes, v0.1.0

# Dev
No dev packages
```

**Comment by izuzak** [comment]
Thanks for the report, @sadlerdavid. Can you share more information about what you were doing when this happened? Are you able to reproduce the problem?

**Comment by sadlerdavid** [comment]
Sorry the the lag on the response @ izuzak.  I believe the fix for this was closing, and re-opening Atom after installing the update.  I have not encountered this problem again.

**Comment by 50Wliu** [comment]
Thanks @sadlerdavid-in that case, I'm going to close this issue.  If you experience it again, please be sure to comment here!

**Comment by izuzak** [comment]
Thanks @sadlerdavid and @50Wliu.

I was just looking into this and noticed a similar but old report: https://github.com/joyent/node/issues/6556. In https://github.com/joyent/node/issues/6556#issuecomment-29897424, it's said that this problem has been fixed in Node 0.11.9.

However, @sadlerdavid s running 0.177.0, which means that he's running the [version which runs on top of io.js](https://github.com/atom/atom/releases/tag/v0.177.0) (which has the same fix mentioned above). As far as I can tell, this is the assertion which is failing:

https://github.com/iojs/io.js/blob/21130c7d6fcaee666f33735768c060be2e06614a/lib/_http_client.js#L309

This sounds like a problem in the TLSSocket implementation in io.js (probably the same one as in Node), so if we manage to reproduce this -- we might report the issue upstream.
